src/Resources/views/AmazonPay/addressBookWidget.html.twig

```diff
@@ -25,19 +21,43 @@
         new OffAmazonPayments.Widgets.AddressBook({
             sellerId: "{{ amazon.merchantId }}",
             onOrderReferenceCreate: function (orderReference) {
-
-                $.post('{{ path('tierperso_sylius_amazon_pay_plugin_order_details') }}', {
-                    orderReferenceId: orderReference.getAmazonOrderReferenceId(),
+                amazonOrderReferenceId = orderReference.getAmazonOrderReferenceId();
+                $.post('{{ path('tierperso_sylius_amazon_pay_plugin_get_order_details') }}', {
+                    orderReferenceId: amazonOrderReferenceId,
                     accessToken: accessToken
                 }).done(function (data) {
                     try {
                         JSON.parse(data);
                     } catch (err) {
                     }
-                    $("#get_details_response").html(data);
+                    console.log(data);
                 });
+
             },
             onAddressSelect: function (orderReference) {
+                $.post('{{ path('tierperso_sylius_amazon_pay_plugin_get_order_details') }}', {
+                    orderReferenceId: amazonOrderReferenceId,
+                    accessToken: accessToken
+                }).done(function (data) {
+                    try {
+                        JSON.parse(data);
+                    } catch (err) {
+                    }
+                    var name = data.GetOrderReferenceDetailsResult.OrderReferenceDetails.Destination.PhysicalDestination.Name;
+                    var splitName = name.split(" ");
```

Comment:
Hm, and what about taking care of the whole of this code and logic in the Backend (controller)? 🤔 YOU ARE a developer, you can do anything you need and want. If it's hard to operate `data.GetOrderReferenceDetailsResult.OrderReferenceDetails.Buyer.*`, maybe make an object that will be returned into Twig template instead of this ugly set of arrays. Don't you think it's a total mess in the code blow? I know it's JS, but it is way too hard to read this at the first glance. *Code must be readable*